prov/rxm/src/rxm_conn.c

```diff
@@ -190,49 +172,60 @@ void *rxm_msg_listener(void *arg)
 
 	FI_DBG(&rxm_prov, FI_LOG_FABRIC, "Starting MSG listener thread\n");
 	while (1) {
-		rd = fi_eq_sread(rxm_fabric->msg_eq, &event, entry, len, -1, 0);
+		rd = fi_eq_sread(rxm_ep->msg_eq, &event, entry, len, -1, 0);
 		/* We would receive more bytes than sizeof *entry during CONNREQ */
 		if (rd < 0) {
 			if (rd == -FI_EAVAIL)
 				OFI_EQ_READERR(&rxm_prov, FI_LOG_FABRIC,
-						rxm_fabric->msg_eq, rd, err_entry);
+						rxm_ep->msg_eq, rd, err_entry);
 			else
 				FI_WARN(&rxm_prov, FI_LOG_FABRIC,
 						"msg: unable to fi_eq_sread\n");
+			if (err_entry.err == ECONNREFUSED) {
+				FI_DBG(&rxm_prov, FI_LOG_FABRIC,
+				       "Connection refused\n");
+				ofi_cmap_process_reject(rxm_ep->util_ep.cmap,
+							(uint64_t)err_entry.fid->context);
+			}
 			continue;
 		}
 
 		switch(event) {
 		case FI_NOTIFY:
-			FI_TRACE(&rxm_prov, FI_LOG_FABRIC, "Closing rxm msg listener\n");
+			FI_TRACE(&rxm_prov, FI_LOG_FABRIC,
+				 "Closing rxm msg listener\n");
 			free(entry);
 			return NULL;
 		case FI_CONNREQ:
 			if (rd != len)
 				goto err;
 			FI_DBG(&rxm_prov, FI_LOG_FABRIC, "Got new connection\n");
-			ret = rxm_msg_process_connreq(entry->fid->context, entry->info,
-					entry->data);
-			if (ret)
-				FI_WARN(&rxm_prov, FI_LOG_FABRIC,
-						"Unable to process connection request\n");
+			cm_data = (void *)entry->data;
+			rxm_msg_process_connreq(rxm_ep, entry->info, entry->data);
 			break;
 		case FI_CONNECTED:
-			FI_DBG(&rxm_prov, FI_LOG_FABRIC, "Connected\n");
-			rxm_msg_process_connect_event(entry->fid, entry->data,
-					rd - sizeof(*entry));
+			FI_DBG(&rxm_prov, FI_LOG_FABRIC,
+			       "Connection successful\n");
+			cm_data = (void *)entry->data;
+			ofi_cmap_process_connect(rxm_ep->util_ep.cmap,
+						 (uint64_t)entry->fid->context,
+						 (rd - sizeof(*entry)) ?
+						 &cm_data->conn_id : NULL);
 			break;
 		case FI_SHUTDOWN:
-			FI_DBG(&rxm_prov, FI_LOG_FABRIC, "Received connection shutdown\n");
-			rxm_msg_process_shutdown_event(entry->fid);
+			FI_DBG(&rxm_prov, FI_LOG_FABRIC,
+			       "Received connection shutdown\n");
+			ofi_cmap_process_shutdown(rxm_ep->util_ep.cmap,
+						  (uint64_t)entry->fid->context);
 			break;
 		default:
-			FI_WARN(&rxm_prov, FI_LOG_FABRIC, "Unknown event: %u\n", event);
+			FI_WARN(&rxm_prov, FI_LOG_FABRIC,
+				"Unknown event: %u\n", event);
 		}
 		continue;
 err:
-		FI_WARN(&rxm_prov, FI_LOG_FABRIC,
-				"Received size (%d) not matching expected (%d)\n", rd, len);
+		FI_WARN(&rxm_prov, FI_LOG_FABRIC, "Received size (%d) not "
+			"matching expected (%d)\n", rd, len);
```

Comment:
I still can't see how it can be reached.  I don't see a 'goto err', and immediately before this label is a continue call.  Maybe it's reached by code not shown in this patch...